### PR TITLE
Embraced test result summary output with information about environment

### LINKED ISSUE #518: Enhance TestResults.txt output with information about used environment
**Summary**
Currently, when student submits Pharo track exercise, submission command produces TestResults.txt which is supplemental inforamtion for mentor, if all tests related to exercise runned successfully. Maintainers would like to have also information about used environment, on which was exercise developed and submitted.  

**Description**
Enhance TestResults.txt with used OS environment (OS name, version), image version, VM version. 
proposed output:
```
Tested on: 2020-04-21 13:36
15 run, 15 passes, 0 skipped, 0 failures, 0 errors.
Used OS/Platform: Mac OS 1000 x86_64
Image version:  Pharo 9.0.0 build.10220 (64 bit)
VM version: v9.0.6 - Commit: 1ee76042
```
**Implementation detail/Additional details**
Modify `TestResult>>exercismSummary` with following details:
OS details
```
		nextPutAll: Smalltalk os platformName; space;
		nextPutAll: Smalltalk os version; space;
		nextPutAll: Smalltalk os subtype; cr
```
Image details:
```
SystemVersion current version.
SystemVersion current build asString. 
Smalltalk vm is32bit 
ifTrue: [ str nextPutAll: ' (32 Bit)' ]
ifFalse: [ str nextPutAll: ' (64 Bit)' ].	
```
VM detals: 
```
Smalltalk vm interpreterSourceVersion 
```

### PR DESCRIPTION
**Pull request by Bajger**
- added information about OS, platform, Image version, VM version
- added unit tests for test result summary content
- fixes #518

### CODE REVIEW

**Comment by Bajger** [comment]
@ErikSchierboom Hi! I don't know what to do with failing linting rule. Is there something to fix in the config files?
@glennj Please review.

**Comment by SleeplessByte** [comment]
```
A `files.exemplar` value is `.meta/solution/%{pascal_slug}.class.st`, 
which is not a unique `files` entry:
/home/runner/work/pharo-smalltalk/pharo-smalltalk/config.json
```

It's complaining about these two being the same:

```json
{
  "example": [
    ".meta/solution/%{pascal_slug}.class.st"
  ],
  "exemplar": [
    ".meta/solution/%{pascal_slug}.class.st"
  ]
}
```

I know there is / was an open issue about this, because this should be possible. In other words: ignore it!

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
@SleeplessByte There is an open issue: https://github.com/exercism/configlet/issues/443

**Comment by Bajger** [comment]
@SleeplessByte Can I just remove `exemplar` (in separate PR) field from config file since it isn't anyhow used at this moment on Pharo Smalltalk track? Or rather to wait for fix on this issue?

**Comment by SleeplessByte** [comment]
Exemplar will be used by concept exercises a d removing it will trigger a different error. 

Ignore and merge :)

File: dev/src/ExercismTools/TestResult.extension.st [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+		crlf;
+		nextPutAll: 'Image version: ';
+		nextPutAll: SystemVersion current version; space;
+		nextPutAll: 'build: ';
+		nextPutAll: SystemVersion current build asString.
+		Smalltalk vm is32bit 
+			ifTrue: [ aStream nextPutAll: ' (32 Bit)' ]
+			ifFalse: [ aStream nextPutAll: ' (64 Bit)' ].
+	aStream 
+		crlf;
```
[the remark this answers is not in the document]
Comment (Bajger) [reply dev/src/ExercismTools/TestResult.extension.st:20]: Yes you're right! Will add commit to simplify that.

[the remark this answers is not in the document]
Comment (Bajger) [reply dev/src/ExercismTools/TestResult.extension.st]: Done.

**Comment by Bajger** [comment]
Review resolved. @glennj Please merge (or whoever has write access to repo)

**Comment by ErikSchierboom** [comment]
Done

**Pull request merged**